core/cmd/local_client.go

```diff
@@ -240,20 +238,57 @@ func (cli *Client) runNode(c *clipkg.Context) error {
 	}
 
 	lggr.Info("API exposed for user ", user.Email)
+
+	grp, grpCtx := errgroup.WithContext(rootCtx)
+
 	if err = app.Start(); err != nil {
 		return errors.Wrap(err, "error starting app")
 	}
-	defer func() {
-		lggr.ErrorIf(app.Stop(), "Error stopping app")
-		if err = lggr.Sync(); err != nil {
-			log.Fatal(err)
+
+	grp.Go(func() error {
+		<-grpCtx.Done()
+		defer releaseDbLock()
+		if err = app.Stop(); err != nil {
+			return errors.Wrap(err, "error stopping app")
 		}
-	}()
+		return nil
+	})
 
 	lggr.Debug("Environment variables\n", config.NewConfigPrinter(cli.Config))
 
 	lggr.Infow(fmt.Sprintf("Chainlink booted in %.2fs", time.Since(static.InitTime).Seconds()), "appID", app.ID())
-	return (cli.Runner.Run(app))
+
+	// After SIGTERM grace period, we must fail fast, but try closing DB
+	go func() {
+		<-rootCtx.Done()
```

Comment:
I wonder if this goroutine should be more directly coupled to the incoming signal. By sharing the rootCtx, other goroutines could be scheduled before this one, delaying the start of the grace period count down. Maybe not significant in practice though :shrug:

Reply:
Good point! I will try addressing this next together with your previous comment.

Reply:
Fixed.